README.md

```diff
@@ -1,141 +1,678 @@
+  - [fromArraySchema](#fromarrayschema)
+  - [objectValidator](#objectvalidator)
+  - [arrayValidator](#arrayvalidator)
+
+### Typechecking
+
+The validator library provides some typechecking helper functions out of the box. They all accept a value and return a boolean.
+
+#### isISOString
+
+Tests whether the value is a valid ISO8601 date string.
+
+```js
+isISOString('2018-07-04'); //=> true
+
+isISOString('tuesday'); //=> false
+```
+
+#### isDate
+
+Tests whether the value is either a date object or a valid (ISO8601) date string.
+
+```js
+isDate(new Date('2017-07-07')); //=> true
+
+isDate(new Array()); //=> false
+```
+
+#### isObject
+
+Tests whether the value is an object (but not an array or date).
+
+```js
+isObject({ x: 1, y: 2 };); //=> true
+
+isObject([1, 2]); //=> false
+```
+
+#### isArray
+
+Tests whether the value is an array.
+
+```js
+isArray(['a', 'b']); //=> true
+
+isArray({ a: 'b' }); //=> false
+```
+
+#### isNull
+
+Tests whether the value is null.
+
+```js
+isNull(null); //=> true
+
+isNull(undefined); //=> false
+```
+
+#### isType
+
+Tests whether the value is of the given type, using `typeof`. This is useful for cases where the library doesn't have a specific function built in.
+
+```js
+const isNumber = isType('number');
+
+isNumber(47); //=> true
+
+isNumber('47'); //=> false
+```
+
+### Default validators
+
+A validator is a function that accepts some data and returns a [Result](#results). Most of the time you will probably want to write your own custom logic or use a [schema](#working-with-schemas), but the library exports some default validators for common use cases.
+
+#### alwaysErr
+
+Creates a validator that always returns the given error. Accepts either an error string or an array of error strings.
+
+```js
+const validate = alwaysErr('Uh oh');
+
+getErrors(validate(5)); //=> ["Uh oh"]
+```
+
+#### alwaysOK
+
+Creates a validator that always succeeds.
+
+```js
+const validate = alwaysOK();
+
+isOK(validate(123)); //=> true
+```
+
+#### fromPredicate
+
+Creates a validator from a predicate function. The first argument is the predicate, which accepts data and returns a boolean. The second is a function that accepts the same data and returns an error string, which will be used if the predicate fails.
+
+```js
+const validate = fromPredicate(
+  n => n >= 3,
+  n => `Validation failed: ${n} was less than 3`
+);
+
+isOK(validate(5)); //=> true
+
+getErrors(validate(2)); //=> ["Validation failed: 2 was less than 3"]
+```
+
+#### validateIsType
+
+Validates that the data is of the specified type.
+
+```js
+const validate = validateIsType('number');
+
+isOK(validate(5)); //=> true
+
+isErr(validate('thirty')); //=> true
+```
+
+#### validateIsIn
+
+Validates that the data is in the specified array, according to the ES6 [`.includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) method.
+
+```js
+const validate = validateIsIn(['a', 'b']);
```

Comment:
Out of curiosity, if I did the following, would it work, or be an error?

```js
const validate = validateIsIn('foobar');

validate('foo'); //=> ????
```

This is given that othings other than arrays support `.includes`

Reply:
I think that'd work, yep